Documented uptodate action in Common Workflows (#3825)
uyuni-project/uyuni-docs: modules/common-workflows/pages/workflow-clients-update-rec-actions.adoc

.. Upgrading the packages to their latest available version by using the client's package manager.
 The version depends on the synchronized repositories in {productname}.
 This is different depending on the operating system of the client:

... For Debian-based clients, such as Debian or Ubuntu, the action executes [command]``apt dist-upgrade -q -y $PACKAGES``.
... For RPM-based clients that are not SUSE, such as Red Hat Enterprise Linux or SUSE Liberty Linux, the action executes [command]``yum --quiet -y update $PACKAGES`` or [command]``dnf --quiet -y upgrade $PACKAGES`` (depending on the package manager the client is using).
... For non-transactional SUSE clients, such as SUSE Linux Enterprise 15, the action executes [command]``zypper --non-interactive --auto-agree-with-licenses update $PACKAGES``.
... For transactional SUSE clients, the action executes the same in a transactional shell.

--

+

// . version 4.3" if SUMULIMA detects that reboot is necessary, for 4.3.x, it will automatically reboot that client.

+

// 5.0 and later

. With [systemitem]``reboot`` and [systemitem]``rebootifneeded`` two reboot strategies exist.

>   Users have the option to select and enable one reboot action from them.

Author: your reasoning makes a lot of sense!

+

--

[systemitem]``rebootifneeded``::

Reboot detection is specific to the client operating system.

* For Debian or Ubuntu, see https://www.debian.org/doc/debian-policy/ch-opersys.html#signaling-that-a-reboot-is-required.

* For non-transactional {suse} clients, {productname} checks if any of the patches require reboot by using [command]``zypper -x list-patches``, and reboots if so.

> * For transactional {suse} clients, {productname} checks if there is a pending transaction, and reboots if so.

Author: yes, I also agree here.

+

> // FIXME: is this meant?

Author: I was unsure about the following statement. You clarified it with the suggestion--thanks!  I'll now remove this FIXME line completely.

In the past, I had trouble with `//` comments between items.  At least sometimes, a comment seems to start a new list.  Maybe, whitespace caused this (but we seem to need these empty line for the tool (po4a?) that creates the translation templates).

+

> * For the Red Hat-based clients, {productname} uses [command]``dnf -q needs-restarting -r``, and reboots if so.

Reviewer: [nit] same as above :) 

```suggestion
* For the Red Hat-based clients, {productname} reboots if [command]``dnf -q needs-restarting -r`` indicates that a reboot is required.
```
